src/coreclr/vm/peimagelayout.cpp

```diff
@@ -299,8 +306,12 @@ void PEImageLayout::ApplyBaseRelocations(bool relocationMustWriteCopy)
             }
         }
 
+#if defined(__APPLE__) && defined(HOST_ARM64)
+        BOOL bExecRegion = true;
+#else
         BOOL bExecRegion = (dwOldProtection & (PAGE_EXECUTE | PAGE_EXECUTE_READ |
             PAGE_EXECUTE_READWRITE | PAGE_EXECUTE_WRITECOPY)) != 0;
+#endif
```

Comment:
I have added this to make sure that the instruction cache gets flushed on Apple Silicon too. I am not sure whether this is required?

Reply:
Agreed. I changed the logic to use `PAL_JitWriteProtect` where the protection is toggled on other platforms to minimise range to flush.